XdistroSetup.sh

```diff
@@ -0,0 +1,205 @@
+#!/bin/bash
+set -eu -o pipefail
+#
+# Script to automate basic preparation of a cross-distro
+# bootstrap-builder host for a given alternate-distro
+# build-target
+#
+#################################################################
+PROGNAME=$( basename "$0" )
+RUNDIR="$( dirname "$0" )"
+DEBUG="${DEBUG:-UNDEF}"
+
+# Make interactive-execution more-verbose unless explicitly told not to
+if [[ $( tty -s ) -eq 0 ]] && [[ ${DEBUG} == "UNDEF" ]]
+then
+   DEBUG="true"
+fi
+
+
+# Error handler function
+function err_exit {
+   local ERRSTR
+   local ISNUM
+   local SCRIPTEXIT
+
+   ERRSTR="${1}"
+   ISNUM='^[0-9]+$'
+   SCRIPTEXIT="${2:-1}"
+
+   if [[ ${DEBUG} == true ]]
+   then
+      # Our output channels
+      logger -i -t "${PROGNAME}" -p kern.crit -s -- "${ERRSTR}"
+   else
+      logger -i -t "${PROGNAME}" -p kern.crit -- "${ERRSTR}"
+   fi
+
+   # Only exit if requested exit is numerical
+   if [[ ${SCRIPTEXIT} =~ ${ISNUM} ]]
+   then
+      exit "${SCRIPTEXIT}"
+   fi
+}
+
+
+# Print out a basic usage message
+function UsageMsg {
+   local SCRIPTEXIT
+   SCRIPTEXIT="${1:-1}"
+
+   (
+      echo "Usage: ${0} [GNU long option] [option] ..."
+      echo "  Options:"
+      printf '\t%-4s%s\n' '-d' 'Distro nickname (e.g., "Rocky")'
+      printf '\t%-4s%s\n' '-h' 'Print this message'
+      printf '\t%-4s%s\n' '-k' 'List of RPM-validation key-files or RPMs'
+      printf '\t%-4s%s\n' '-r' 'List of repository-related RPMs'
+      echo "  GNU long options:"
+      printf '\t%-20s%s\n' '--distro-name' 'See "-d" short-option'
+      printf '\t%-20s%s\n' '--help' 'See "-h" short-option'
+      printf '\t%-20s%s\n' '--repo-rpms' 'See "-r" short-option'
+      printf '\t%-20s%s\n' '--sign-keys' 'See "-k" short-option'
+   )
+   exit "${SCRIPTEXIT}"
+}
+
+# Install the alt-distro's GPG key(s)
+function InstallGpgKeys {
+   local LOOP
+   LOOP=0
+
+   while [[ LOOP -lt ${#PKGSIGNKEYS[*]} ]]
+   do
+      if [[ ${PKGSIGNKEYS[${LOOP}]} == "" ]] 
+      then
+         break
+      elif [[ ${PKGSIGNKEYS[${LOOP}]} == *.rpm ]]
+      then
+         echo yum install -y "${PKGSIGNKEYS}"
+      else
+	     printf "Installing %s to /etc/pki/rpm-gpg... " \
+           "${PKGSIGNKEYS[${LOOP}]}"
+         cd /etc/pki/rpm-gpg || err_exit "Could not chdir"
+         curl -sOkL "${PKGSIGNKEYS[${LOOP}]}" || err_exit "Download failed"
+         echo "Success"
+         cd "${RUNDIR}"
+      fi
+      LOOP=$(( LOOP + 1))
+   done
+}
+
+function StageDistroRpms {
```

Comment:
Why do we need this at all? The readme passes the rpms to `OSpackages.sh` with the `-r` option, but that option accepts urls just fine.

Reply:
Mostly, facilitates snarfing the packages before trying to install them so that, if you run into issues (like CentOS packaging its GPG keys in an RPM but Alma and Rocky no), it's easier the relevant RPMs' contents

Reply:
I'd like to just leave it, for now